report_dynamic/models/report_dynamic.py

```diff
@@ -0,0 +1,271 @@
+        compute="_compute_resource_ref",
+        inverse="_inverse_resource_ref",
+        store=True,
+    )
+    wrapper_report_id = fields.Many2one("ir.ui.view", domain="[('type', '=', 'qweb')]")
+    template_id = fields.Many2one(
+        "report.dynamic",
+        domain="[('is_template', '=', True)]",
+        copy=False,
+        default=lambda self: self.env.company.external_report_layout_id,
+    )
+    report_ids = fields.One2many(
+        "report.dynamic",
+        "template_id",
+        domain="[('is_template', '=', False)]",
+        copy=False,
+    )
+    documentation = fields.Text(default="Documentation placeholder", readonly=True)
+    condition_domain_global = fields.Char(
+        string="Global domain condition", default="[]"
+    )
+    active = fields.Boolean(default=True)
+    is_template = fields.Boolean(default=False)
+    lock_date = fields.Date()
+    field_ids = fields.Many2many(
+        "ir.model.fields", "contextual_field_rel", "contextual_id", "field_id", "Fields"
+    )
+    window_action_exists = fields.Boolean(compute="_compute_window_action_exists")
+    group_by_record_name = fields.Char(
+        compute="_compute_group_by_record_name",
+        store=True,
+        help="Computed field for grouping by record name in search view",
+    )
+
+    @api.model
+    def _selection_target_model(self):
+        models = self.env["ir.model"].search([])
+        return [(model.model, model.name) for model in models]
+
+    @api.depends("model_id", "res_id", "template_id")
+    def _compute_resource_ref(self):
+        for this in self:
+            model = (
+                this.model_id.model
+                if this.is_template
+                else this.template_id.model_id.model
+            )
+            if model:
+                # Return a meaningful message anytime this breaks
+                try:
+                    sample_record = self.env[model].search([], limit=1)
+                except Exception as e:
+                    raise UserError(
+                        _("Model {} is not applicable for report. Reason: {}").format(
+                            model, str(e)
+                        )
+                    )
+                # Tackle the problem of non-existing sample record
+                if not sample_record:
+                    raise UserError(
+                        _(
+                            "No sample record exists for Model {}. "
+                            "Please create one before proceeding"
+                        ).format(model)
+                    )
+                # we need to give a default to id part of resource_ref
+                # otherwise it is not editable
+                this.resource_ref = "{},{}".format(
+                    model, this.res_id or sample_record.id,
+                )
+            else:
+                this.resource_ref = False
+
+    def _inverse_resource_ref(self):
+        for this in self:
```

Comment:
```suggestion
        for rec in self:
```
please do not use "this", is not JS :wink: